src/adapter/src/coord.rs

```diff
@@ -788,10 +783,39 @@ impl<S: Append + 'static> Coordinator<S> {
         }
     }
 
-    #[allow(dead_code)]
-    async fn create_storage_export(&mut self, id: GlobalId, sink: &Sink) {
+    async fn create_storage_export(
+        &mut self,
+        id: GlobalId,
+        sink: &Sink,
+        connection: StorageSinkConnection,
+    ) -> Result<(), AdapterError> {
+        // Validate `sink.from` is in fact a storage collection
+        self.controller.storage().collection(sink.from)?;
+
+        // The AsOf is used to determine at what time to snapshot reading from the persist collection.  This is
+        // primarily relevant when we do _not_ want to include the snapshot in the sink.  Choosing now will mean
+        // that only things going forward are exported.
+        let timeline = self
+            .get_timeline(sink.from)
+            .unwrap_or(Timeline::EpochMilliseconds);
```

Comment:
What would happen here for sink that created from a constant query like `SELECT 1`? It may not belong to any timeline in which case I'm not sure what the correct `as_of` would be.

Reply:
Yeah that sounds correct to me.